static/nav-highlight.js

```diff
@@ -0,0 +1,54 @@
+// Highlight sidebar nav items based on the current scroll position.
+
+// Number of category nav items on the page
+const numCategories = document.getElementById('organized_list_items');
+const num = Number(numCategories.dataset.name);
+
+var timer = null;
+let currentElem = 1;
+
+// Find and highlight the currently scrolled section
+const findHiddenElem = () => {
+  let elemIndex = 0;
+  for (let i = 0; i < num; i += 1) {
+    const elem = document.getElementsByClassName('organized-by')[i].getBoundingClientRect();
+    if (elem.top < 60) {
+      elemIndex += 1;
+    }
+  }
+  elemIndex += 1;
+  if(currentElem !== elemIndex) {
+    document.body
+      .querySelector('.table tbody tr:nth-child(' + currentElem + ') a')
+      .removeAttribute("style");
+  }
+  document.body
+    .querySelector('.table tbody tr:nth-child(' + elemIndex + ') a')
+    .setAttribute("style", "color: #576523; font-weight: bold;");
+  currentElem = elemIndex;
+}
+
+// Run the callback `fn` at most every `ms`
+function debounce (ms, fn) {
+  let timeout;
+  return function () {
+    const args = arguments;
+    const later = () => {
+      timeout = null;
+      fn.apply(null, args);
+    };
+    clearTimeout(timeout);
+    timeout = setTimeout(later, ms);
+  }
+}
+
+
+// On pageload and on scroll (with a 100ms delay), find the current section the
+// user has scrolled to and highlight that in the navigation sidebar
+findHiddenElem();
```

Comment:
Before you had this in a "load" event listener, but you were calling the function immediately, so it was actually being called on the event. But in any case you don't need to use a load event listener. Since the script tag is at the bottom of the body, we know the dom is loaded before it runs.